File: PracticaE02/PracticaE02Back/src/main/java/com/softtek/practicae02back/modelo/Conductor.java
package com.softtek.practicae02back.modelo;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
@Entity
@Table(name = "Conductores")
public class Conductor {
    @Id
    @Column(length = 10)
    private String dni;
    @Column(length = 60)
    private String nombre;
}

File: PracticaE02/PracticaE02Front/src/app/_model/conductor.ts
export type HTTPConductorResponse = {dni: string, nombre: string};
export class Conductor {

    private _dni: string;
    public get dni(): string {
        return this._dni;
    }
    public set dni(value: string) {
        this._dni = value;
    }
    private _nombre: string;
    public get nombre(): string {
        return this._nombre;
    }
    public set nombre(value: string) {
        this._nombre = value;
    }

    constructor(dni: string, nombre: string) {
        this._dni = dni;
        this._nombre = nombre;
    }

    public static fromHttpResponse(obj: HTTPConductorResponse): Conductor {
        return new Conductor(obj.dni, obj.nombre);
    }
    
}